### PR TITLE
PrivateCodeHost: AWS Private Link

### PR DESCRIPTION
**Pull request by filiphaftek**
[issue](https://github.com/sourcegraph/customer/issues/1858)

Contains:
- AWS Private Link (VPC Endpoint Service) configuration playbook for customer
- VPC Endpoint configuration for code in mi2 controller

### CODE REVIEW

**Review by unknwon — APPROVED** [review]
Can't validate the content yet, but always love seeing people writing docs!

File: content/departments/cloud/technical-docs/private-code-hosts/index.md [hunk 1]
```diff
@@ -66,6 +66,58 @@ For more details, go to [Google documentation](https://cloud.google.com/network-
 For each customer using private code host, additional section to our generated operation dashboard is added.
 Upon enabling the private code host support, follow the process [to update the dashboard](https://github.com/sourcegraph/cloud/blob/main/prod.dashboard.md#update-all-generated-dashboards)
 
+### AWS VPC Endpoint Service (aka AWS Private Link)
+
+[AWS Private Link](https://aws.amazon.com/privatelink/) is secure method to expose single endpoint (i.e. code host) from AWS VPC to specific AWS Principals (IAM in specific AWS VPC).
```
Comment (evict) [inline content/departments/cloud/technical-docs/private-code-hosts/index.md]:
```suggestion
[AWS Private Link](https://aws.amazon.com/privatelink/) is a secure method to expose a single endpoint (i.e. code host) from AWS VPC to specific AWS Principals (IAM in specific AWS VPC).
```

File: content/departments/cloud/technical-docs/private-code-hosts/index.md [hunk 2]
```diff
@@ -66,6 +66,58 @@ For more details, go to [Google documentation](https://cloud.google.com/network-
...
+This solution ensures code host is not exposed from AWS and only given principal can access it.
```
Comment (evict) [inline content/departments/cloud/technical-docs/private-code-hosts/index.md]:
```suggestion
This solution ensures code the host is not exposed from AWS and only a given principal can access it.
```

File: content/departments/cloud/technical-docs/private-code-hosts/index.md [hunk 3]
```diff
@@ -66,6 +66,58 @@ For more details, go to [Google documentation](https://cloud.google.com/network-
...
+
+When customer exposes code host via VPC Endpoint Service, Cloud Instance needs to create [AWS VPC Endpoint](https://docs.aws.amazon.com/whitepapers/latest/aws-privatelink/what-are-vpc-endpoints.html) to expose connection inside Sourcegraph Managed AWS VPC.
```
Comment (evict) [inline content/departments/cloud/technical-docs/private-code-hosts/index.md]:
```suggestion
When a customer exposes the code host via VPC Endpoint Service, the Cloud Instance needs to create an [AWS VPC Endpoint](https://docs.aws.amazon.com/whitepapers/latest/aws-privatelink/what-are-vpc-endpoints.html) to expose connections inside the Sourcegraph Managed AWS VPC.
```

File: content/departments/cloud/technical-docs/private-code-hosts/index.md [hunk 4]
```diff
@@ -109,6 +161,53 @@ cd kubernetes
 kustomize build --load-restrictor LoadRestrictionsNone --enable-helm . | kubectl apply -f -
 ```
 
-### AWS Private Link playbook
+### AWS Private Link playbook for customer
+
+When customer has private code host inside AWS VPC and needs to expose it for Sourcegraph managed AWS VPC, customer can use [AWS Documentation](https://docs.aws.amazon.com/vpc/latest/privatelink/create-endpoint-service.html) or follow the steps:
```
Comment (evict) [inline content/departments/cloud/technical-docs/private-code-hosts/index.md]:
```suggestion
When a customer has private code hosts inside the AWS VPC and needs to expose it for Sourcegraph managed AWS VPC, customers can use [AWS Documentation](https://docs.aws.amazon.com/vpc/latest/privatelink/create-endpoint-service.html) or follow these steps:
```

**Review by evict — APPROVED** [review]

**Review by evict — COMMENTED** [review]
I would recommend maybe drawing a visual aid? So it's easier to visualize the concept.

**Pull request merged**